### PR TITLE
[lint] Add lint targets and waiver files for remaining IPs

### PR DESCRIPTION
**Pull request by msfschaffner**
This is a follow-up PR of #1304 and basically replicates what has been done in that PR for all other IPs. Further, this PR adds all the ascentlint and verilator waiver files.

Do not merge this before the precursor PR has been merged.

### CODE REVIEW

**Comment by msfschaffner** [comment]
Rebased after merging in precursor PRs.

File: hw/vendor/lowrisc_ibex/ibex_core.core [hunk 1]
```diff
@@ -69,9 +69,6 @@ targets:
 
   lint:
     filesets:
-      # Note on Verilator waivers:
-      # You *must* include the waiver file first, otherwise only global waivers
-      # are applied, but not file-specific waivers.
       - tool_verilator ? (files_lint_verilator)
```
Comment (imphil) [inline hw/vendor/lowrisc_ibex/ibex_core.core]: Please don't change code in the vendor directory, this needs to go through Ibex.

Comment (msfschaffner) [reply hw/vendor/lowrisc_ibex/ibex_core.core]: Sure, will upstream to Ibex.

**Comment by msfschaffner** [comment]
Removed the change in `ibex_core.core`.

**Pull request merged**